### ISSUE TITLE
Web App Class Attendance

### ISSUE THREAD

**Issue by Scott123180** [Capping-CPCA/Database#56]
@marcosb1 the thing we discussed last night
Just need two things for class attendance
- Web app new jail person function: A function that takes first, middle, last, race, age, number of children under 18 and zip code (_The one in ParticipantClassAttendance_) and returns the ID of this new person when he/she is created.
- Web App class submission proc: A proc for submitting attendance that takes an array of participant ID's, all of the ClassOffering information, the facilitator ID. It should make a new instance of a ClassOffering, many instances of ParticpantClassAttendance, and one instance of FacilitatorClassAttendance.
:v:

**Comment by marcosb1** [comment]
@Scott123180 
Regarding the first request, we can have that done no problem.
However, the second request I believe is a little out-of-scope. Passing an "array" of participantIDs into a postgres function is never a good idea. Ideally how it should work is, you write a function in php for this that calls several stored procedures (Create ClassOffering --> Facilitator Attendance Insert --> Loop through the participants and call the Participant Attendance Insert). Of course plgsql is complete with for loops and recursion, but this does not mean it should be used lightly. Thus I am going to call the second one out of scope. Sorry.

**Comment by Scott123180** [comment]
Thanks for the response. Sounds reasonable, I'm not too familiar with pgSQL so I wasn't sure how feasible it was to have them all in one.

I don't think that there are procs for creating a ClassOffering and FacilitatorClassAttendance so should I open separate issues for those?

I looked at the particpantAttendanceInsert and there's a lot of duplicated data that I don't need if I have the ID and class information for a participant. Also a lot of the information needed for that proc isn't included in the view where I get previous people's information from. I think the cleaner thing to do might be to make a proc that takes in just the table information for participantClassAttendance and the participantID (which I'll have because of the lookup and all of the information returned from the function for a new person).

**Comment by marcosb1** [comment]
Resolved in 28051dde43f7545c06aca56b24e59be3ecd6e515.
I had to make the stored procedure in the following way because of the fields that are required to make an outOfHouse participant. Let us know if there are any issues. I believe the matter is closed, I've tested the procedure and it works how it should.
```sql
SELECT createOutOfHouseParticipant(
   participantFirstName := 'PARTICIPANT_FIRST_NAME'::TEXT,
   participantMiddleInit := 'PARTICIPANT_MIDDLE_INITIAL'::TEXT,
   participantLastName := 'PARTICIPANT_LAST_NAME'::TEXT,
   participantAge := PARTICIPANT_AGE::INT,
   participantRace := 'PARTICIPANT_RACE'::RACE,
   participantSex := 'PARTICIPANT_SEX'::SEX,
   participantDescription := 'PARTICIPANT_DESCRIPTION'::TEXT)
```

Please see the [wiki](https://github.com/Capping-CPCA/Database/blob/master/doc/users/UsingStoredProcedures.md) for sample PHP code as well.
